sections/futures/UserInfo/UserInfo.tsx

```diff
@@ -43,26 +46,33 @@ const FutureTabs = Object.values(FuturesTab);
 
 const UserInfo: React.FC = () => {
 	const router = useRouter();
-	const { walletAddress } = Connector.useContainer();
+	const dispatch = useAppDispatch();
 
 	const marketAsset = useAppSelector(selectMarketAsset);
 	const position = useAppSelector(selectPosition);
+	const walletAddress = useAppSelector(selectWallet);
+	const statues = useAppSelector(selectQueryStatuses);
+	const tradesQuery = statues.trades;
 
 	const openOrders = useAppSelector(selectOpenOrders);
 	const accountType = useAppSelector(selectFuturesType);
+	const trades = useAppSelector(selectUsersTradesForMarket);
+
+	useFetchAction(fetchTradesForSelectedMarket, {
+		dependencies: [walletAddress, accountType, position?.position?.size.toString()],
+		disabled: !walletAddress,
+	});
 
 	const [showShareModal, setShowShareModal] = useState(false);
 	const [hasOpenPosition, setHasOpenPosition] = useState(false);
 	const [openProfitCalcModal, setOpenProfitCalcModal] = useState(false);
 
+	// TODO: Move to sdk / redux
 	const marginTransfersQuery = useGetFuturesMarginTransfers();
-	const marginTransfers = useMemo(() => marginTransfersQuery?.data ?? [], [
-		marginTransfersQuery.data,
-	]);
-
-	const futuresTradesQuery = useGetFuturesTradesForAccount(marketAsset, walletAddress);
-
-	const history = useMemo(() => futuresTradesQuery?.data ?? [], [futuresTradesQuery.data]);
+	const marginTransfers = useMemo(
+		() => (marginTransfersQuery.isSuccess ? marginTransfersQuery?.data ?? [] : []),
+		[marginTransfersQuery.isSuccess, marginTransfersQuery.data]
+	);
```

Comment:
This is equivalent to:
```suggestion
		() => (marginTransfersQuery?.data ?? []),
		[marginTransfersQuery.data]
	);
```